rhino/src/test/java/org/mozilla/javascript/tests/ContextFactoryFeatureTest.java

```diff
@@ -0,0 +1,78 @@
+package org.mozilla.javascript.tests;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.mozilla.javascript.Context;
+import org.mozilla.javascript.ContextFactory;
+
+public class ContextFactoryFeatureTest {
+    @BeforeAll
+    public static void cleanup() {
+        // Not all the OTHER tests currently clean up their context, and
+        // this test can't work unless they do.
+        Context cx = Context.getCurrentContext();
+        if (cx != null) {
+            Context.exit();
+        }
+    }
+
+    @Test
+    public void defaultContextFactory() {
+        ContextFactory factory = new ContextFactory();
+        try (Context cx = factory.enterContext()) {
+            assertTrue(cx.hasFeature(Context.FEATURE_V8_EXTENSIONS));
+            assertTrue(cx.hasFeature(Context.FEATURE_ENABLE_XML_SECURE_PARSING));
+            assertFalse(cx.hasFeature(Context.FEATURE_LITTLE_ENDIAN));
+            assertFalse(cx.hasFeature(Context.FEATURE_DYNAMIC_SCOPE));
+        }
+    }
```

Comment:
Maybe we should have a similar test case for every language version and just check if we have to correct feature setup.